### ISSUE TITLE
Example Tests

### ISSUE THREAD

**Issue by BenjamenMeyer** [oauth2-proxy/mockoidc#52]
It would be extremely helpful to have an example unit test and/or integration test to see how this all fits together.

**Comment by rendicott** [comment]
yes, agreed. Took me about 4 hrs of trial and error to get it working. I have grand plans of opening a PR with examples. Problem is that it's specific to my use case and internal company oauth provider so it's not universal.
